vendor/github.com/containers/image/v5/docker/docker_transport.go

```
@@ -43,30 +43,44 @@ func (t dockerTransport) ValidatePolicyConfigurationScope(scope string) error {
 type dockerReference struct {
-	ref reference.Named // By construction we know that !reference.IsNameOnly(ref)
+	ref             reference.Named // By construction we know that !reference.IsNameOnly(ref)
+	isUnknownDigest bool
 }
 
 // ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an Docker ImageReference.
 func ParseReference(refString string) (types.ImageReference, error) {
 	if !strings.HasPrefix(refString, "//") {
 		return nil, fmt.Errorf("docker: image reference %s does not start with //", refString)
 	}
+	// Check if ref has UnknownDigestSuffix suffixed to it
+	unknownDigest := false
+	if strings.HasSuffix(refString, reference.UnknownDigestSuffix) {
+		unknownDigest = true
+		refString = strings.TrimSuffix(refString, reference.UnknownDigestSuffix)
+	}
 	ref, err := reference.ParseNormalizedNamed(strings.TrimPrefix(refString, "//"))
 	if err != nil {
 		return nil, err
 	}
+
 	ref = reference.TagNameOnly(ref)
-	return NewReference(ref)
+	return newReference(ref, unknownDigest)
 }
 
 // NewReference returns a Docker reference for a named reference. The reference must satisfy !reference.IsNameOnly().
 func NewReference(ref reference.Named) (types.ImageReference, error) {
-	return newReference(ref)
+	return newReference(ref, false)
+}
+
+// NewReferenceUnknownDigest returns a Docker reference for a named reference taking into account
+// the value of unknownDigest. The reference doesn't have to satisfy !reference.IsNameOnly()
+func NewReferenceUnknownDigest(ref reference.Named, unknownDigest bool) (types.ImageReference, error) {
+	return newReference(ref, unknownDigest)
 }
 
 // newReference returns a dockerReference for a named reference.
-func newReference(ref reference.Named) (dockerReference, error) {
-	if reference.IsNameOnly(ref) {
+func newReference(ref reference.Named, unknownDigest bool) (dockerReference, error) {
+	if reference.IsNameOnly(ref) && !unknownDigest {
```

I think this should also fail if `unknownDigest && !IsNameOnly`.

---

Fixed